rapid7/metasploit-framework
# Add auxiliary module for unauth SQLi in Ultimate Member plugin (CVE-2024-1071)

Hello Metasploit Team,

This PR adds a new auxiliary module to exploit an unauthenticated SQL injection vulnerability (CVE-2024-1071) in the **Ultimate Member** plugin for WordPress versions **2.1.3 to 2.8.2**. The vulnerability allows an unauthenticated attacker to extract sensitive data via the `sorting` parameter. With over **200,000 active installations**, this vulnerability poses a significant risk.

The module implements time-based blind SQL injection and retrieves user credentials from the `wp_users` table.

**Verification steps:**

1. Install the vulnerable version of the Ultimate Member plugin (2.8.2 or below).
2. Start `msfconsole` and load the module:
   ```bash
   use auxiliary/scanner/http/wp_ultimate_member_sorting_sqli
   ```
3. Set the target options:
   ```bash
   set RHOSTS <target_ip>
   set TARGETURI /
   ```
4. Run the module:
   ```bash
   run
   ```
5. The module should successfully extract user credentials from the WordPress instance.

**Options:**
- `DIR_ID_MIN` and `DIR_ID_MAX`: Define the range for bruteforcing `directory_id` during the SQLi attack.
- `COUNT`: Specify the number of rows to retrieve from the database.
